Utilities/Cleanup-AzureStorageAccounts.ps1

```diff
@@ -0,0 +1,260 @@
+                        $SkippedDisks += $blob.Length
+                        continue;
+                    }
+                }
+                if ($CleanupAgeInDays -ne $null) {
+                    if ( $FileAge -le $CleanupAgeInDays) {
+                        if ($ShowSkippedLogs) { Write-LogInfo "[$blobCounter/$($blobs.Count)]. $DiskURI : Skipped (File Age  ($FileAge) <= $CleanupAgeInDays)" }
+                        $SkippedDisks += $blob.Length
+                        continue;
+                    } else {
+                        $DeleteAgeString = "$FileAge days old"
+                    }
+                } else {
+                    $DeleteAgeString = $null
+                }
+                if ($Patterns -and !$isBootDiagContainer) {
+                    if ( -not (Check-Pattern -DiskName $blob.Name -Pattern $Patterns) ) {
+                        if ($ShowSkippedLogs) { Write-LogInfo "[$blobCounter/$($blobs.Count)]. $DiskURI : Skipped (Pattern '$Patterns' not found)" }
+                        $SkippedDisks += $blob.Length
+                        continue;
+                    }
+                }
+                if ( -not ($blob.ICloudBlob.Properties.LeaseStatus -eq 'Unlocked')) {
+                    if ($ShowSkippedLogs) { Write-LogInfo "[$blobCounter/$($blobs.Count)]. $DiskURI : Skipped (Locked)" }
+                    $SkippedDisks += $blob.Length
+                    continue;
+                }
+                $Counter += 1
+                $AllDisks += $blob
+                $CleanedDisks += $blob.Length
+                if (-not $DryRun) {
+                    Write-LogInfo "[$blobCounter/$($blobs.Count)]. [Deleted Files=$Counter]. Deleting $DeleteAgeString unattached VHD with Uri: $($blob.ICloudBlob.Uri.AbsoluteUri)"
+                    $blob | Remove-AzStorageBlob -Force
+                    if ($isBootDiagContainer) {
+                        $removeContainer = $true
+                    }
+                } else {
+                    Write-LogInfo "[DryRun] [$blobCounter/$($blobs.Count)]. [Deleted Files=$Counter]. Deleting $DeleteAgeString unattached VHD with Uri: $($blob.ICloudBlob.Uri.AbsoluteUri)"
+                }
+            }
+            if (-not $DryRun) {
+                $blobs = Get-AzStorageBlob -Container $container.Name -Context $context
+                if ($isBootDiagContainer -and $blobs.Count -eq 0 -and $removeContainer) {
+                    Write-LogInfo "Removing empty container $($container.Name)"
+                    Remove-AzStorageContainer -Name  $container.Name -Force -Context $context
+                }
+            }
+        }
+    }
+    Write-LogInfo "Creating CSV file '$SubID.csv'..."
+    $AllDiskObject | Export-Csv -Path "$SubID.csv"
+    Write-LogInfo "Done."
+} catch {
+    Write-LogErr "Exception in Cleanup-AzureStorageAccount.ps1."
```

Comment:
It should include the source line number, and file name from the system. You put hard coded script file name.

Reply:
Done

Reply:
un-resolve conversation, please check your pushed changes

Reply:
The latest changes adds "Managed Disk" cleanup as well.